### DISCUSSION TITLE
behavior "isTouched"

### DISCUSSION THREAD

**Discussion by uyig** [react-hook-form/react-hook-form#8474] [Q&A] [answered]
Hello, tell me please, how you can do it. There is a form, "submit" comes from outside. The fields have an error display check, "isTouched" should be "true", and "error" should not be "undefined". When I click the "Outside Submit" button and send the "submit" event, the "firstName" field "fieldState" has "isTouched" "false", and "error" has an error message, but I can't display the error due to "isTouched" "false".

I expect "isTouched" to change to "true" for wrong fields when it happens "submit".

Sandbox with my case - https://codesandbox.io/s/react-hook-form-js-forked-6z5xr8?file=/src/App.js

And another small question, why does "isTouched" change only when "onBlur"? What about controllers that don't have this event? Some complex component integrated into the form.

Thanks!

**Comment by bluebill1049** [comment, marked as answer]
yes touched only update on `onBlur`, I believe this is pretty much a standard now which means users have interacted with the input. 

> What about controllers that don't have this event?

it does, you just need to plug in the `onBlur` event with your input
